mungegithub/github/github.go

```diff
@@ -1692,3 +1696,65 @@ func (config *Config) ListAllIssues(listOpts *github.IssueListByRepoOptions) ([]
 	}
 	return allIssues, nil
 }
+
+// GetLabels grabs all labels from a particular repository so you don't have to
+// worry about paging.
+func (config *Config) GetLabels() ([]*github.Label, error) {
+	var listOpts github.ListOptions
+	var allLabels []*github.Label
+	page := 1
+	for {
+		glog.V(4).Infof("Fetching page %d of labels", page)
+		listOpts = github.ListOptions{PerPage: 100, Page: page}
+		labels, response, err := config.client.Issues.ListLabels(config.Org, config.Project, &listOpts)
+		config.analytics.ListLabels.Call(config, response)
+		if err != nil {
+			return nil, err
+		}
+		for i := range labels {
+			label := &labels[i]
+			allLabels = append(allLabels, label)
+		}
+		if response.LastPage == 0 || response.LastPage <= page {
+			break
+		}
+		page++
+	}
+	return allLabels, nil
+}
+
+// AddLabel adds a single github label to the repository.
+func (config *Config) AddLabel(label *github.Label) error {
```

Comment:
I much prefer the analytics be collected and we print a log message for any 'write' type operation before the `DryRun` return.